src/chainhooks/mod.rs

```diff
@@ -0,0 +1,81 @@
+use std::path::PathBuf;
+
+pub mod types;
+use crate::chainhooks::types::ChainhookSpecificationFile;
+
+use orchestra_event_observer::chainhooks::types::{ChainhookSpecification, HookFormation};
+
+use orchestra_types::{BitcoinNetwork, StacksNetwork};
+
+use std::fs;
+
+pub fn load_chainhooks(
+    manifest_path: &PathBuf,
+    networks: &(BitcoinNetwork, StacksNetwork),
+) -> Result<HookFormation, String> {
+    let hook_files = get_chainhooks_files(manifest_path)?;
+    let mut stacks_chainhooks = vec![];
+    let mut bitcoin_chainhooks = vec![];
+    for (path, relative_path) in hook_files.into_iter() {
+        match ChainhookSpecificationFile::parse(&path, networks) {
+            Ok(hook) => match hook {
+                ChainhookSpecification::Bitcoin(hook) => bitcoin_chainhooks.push(hook),
+                ChainhookSpecification::Stacks(hook) => stacks_chainhooks.push(hook),
+            },
+            Err(msg) => return Err(format!("{} syntax incorrect: {}", relative_path, msg)),
+        };
+    }
+    Ok(HookFormation {
+        stacks_chainhooks,
+        bitcoin_chainhooks,
+    })
+}
+
+pub fn check_chainhooks(manifest_path: &PathBuf, output_json: bool) -> Result<(), String> {
+    let hook_files = get_chainhooks_files(manifest_path)?;
+    for (path, relative_path) in hook_files.into_iter() {
+        let _hook = match ChainhookSpecificationFile::parse(
+            &path,
+            &(BitcoinNetwork::Regtest, StacksNetwork::Devnet),
+        ) {
+            Ok(hook) => hook,
+            Err(msg) => {
+                println!("{} {} syntax incorrect\n{}", red!("x"), relative_path, msg);
+                continue;
+            }
+        };
+        println!("{} {} succesfully checked", green!("✔"), relative_path);
+        if output_json {
+            let body = serde_json::to_string_pretty(&_hook).unwrap();
+            println!("{}", body);
+        }
+    }
+    Ok(())
+}
+
+fn get_chainhooks_files(manifest_path: &PathBuf) -> Result<Vec<(PathBuf, String)>, String> {
+    let mut hooks_home = manifest_path.clone();
+    hooks_home.pop();
+    let suffix_len = hooks_home.to_str().unwrap().len() + 1;
```

Comment:
Should this be `prefix_len`?

Reply:
Addressed in https://github.com/hirosystems/clarinet/pull/402/commits/ce821313ed66b9190826dd147cdcc477cf6908f0